src/Tools/CoreFx.Tools/GenerateDepsJson.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public partial class GenerateDepsJson : BuildTask
+    {
+        [Required]
+        public string DepsJsonPath { get; set; }
+
+        public override bool Execute()
```

Comment:
Did you investigate whether there is any code that can be re-used from the `Microsoft.Extensions.DependencyModel` package from core-setup?